test/test.ts

```diff
@@ -0,0 +1,272 @@
+                      }
+                    ],
+                    compareTo: 13,
+                    operator: ConditionalOperator.LT
+                  }
+                },
+                {
+                  query: {
+                    path: [
+                      {
+                        property: "roo"
+                      }
+                    ],
+                    compareTo: 130,
+                    operator: ConditionalOperator.GT
+                  }
+                }
+              ]
+            },
+            {
+              property: "goo"
+            }
+          ]
+        }
+      },
+      {
+        statement: "$.foo.bar[*] ? (@.zoo[1] < 13).roo ? (@ > 130)",
+        expected: {
+          lhs: [
+            {
+              property: "foo",
+            },
+            {
+              array: "bar",
+              element: "*",
+              filterChain: [
+                {
+                  query: {
+                    path: [
+                      {
+                        array: "zoo",
+                        element: 1
+                      }
+                    ],
+                    compareTo: 13,
+                    operator: ConditionalOperator.LT
+                  }
+                }
+              ]
+            },
+            {
+              property: "roo",
+              filterChain: [
+                {
+                  query: {
+                    path: [],
+                    operator: ConditionalOperator.GT,
+                    compareTo: 130
+                  }
+                }
+              ]
+            }
+          ]
+        }
+      },
+      {
+        statement: "$.foo ? (exists(@.bar[*] ? (@.zoo > 130))).bar.size()",
+        expected: {
+          lhs: [
+            {
+              property: "foo",
+              filterChain: [
+                {
+                  query: {
+                    method: "exists",
+                    path: [
+                      {
+                        array: "bar",
+                        element: "*",
+                        filterChain: [
+                          {
+                            query: {
+                              path: [
+                                {
+                                  property: "zoo"
+                                }
+                              ],
+                              operator: ConditionalOperator.GT,
+                              compareTo: 130
+                            }
+                          }
+                        ]
+                      }
+                    ]
+                  }
+                }
+              ]
+            },
+            {
+              property: "bar"
+            },
+            {
+              method: MethodName.SIZE,
+              arguments: []
+            }
+          ]
+        }
+      }
+    ]
+
+    const methodNames = [
+      "type", "size", "double", "ceiling", "floor", "abs", "datetime"
```

Comment:
There's also the method `keyvalue()`

Reply:
I think it's in the parser, I must have left it out of the test. I'll add it in before merging.